adapters/repos/db/repo.go

```diff
@@ -232,10 +232,26 @@ func (db *DB) IndexExists(className schema.ClassName) bool {
 	return db.GetIndex(className) != nil
 }
 
-// GetIndexForIncoming returns the index if it exists or nil if it doesn't
+// TODO-RAFT: Because of interfaces and import order we can't have this function just return the same index interface
+// for both sharding and replica usage. With a refactor of the interfaces this can be done and we can remode the
+// deduplication
+
+// GetIndexForIncomingSharding returns the index if it exists or nil if it doesn't
+// by default it will retry 3 times between 0-150 ms to get the index
+// to handle the eventual consistency.
+func (db *DB) GetIndexForIncomingSharding(className schema.ClassName) sharding.RemoteIndexIncomingRepo {
+	index := db.GetIndex(className)
+	if index == nil {
+		return nil
+	}
+
+	return index
+}
+
+// GetIndexForIncomingReplica returns the index if it exists or nil if it doesn't
```

Comment:
**question:** isn't this func the same as `GetIndexForIncomingReplica` ?

Reply:
See the TODO-RAFT I added above the two